UnitTests/TestWebSockets.cpp

```diff
@@ -491,6 +493,60 @@ TEST_F(TestWebSocket, Bip15X)
    doTest(kTestTcpHost, kTestTcpPort, kTestTcpHost, kTestTcpPort, FirstStart::Server);
 }
 
+TEST_F(TestWebSocket, SslConnectionPlain)
+{
+   server_ = std::make_unique<SslServerConnection>(StaticLogger::loggerPtr, SslServerConnectionParams{});
+   client_ = std::make_unique<SslDataConnection>(StaticLogger::loggerPtr, SslDataConnectionParams{});
+   doTest(kTestTcpHost, kTestTcpPort, kTestTcpHost, kTestTcpPort, FirstStart::Server);
+}
+
+TEST_F(TestWebSocket, SslConnectionSelfSigned)
+{
+   auto privKeyServer = bs::network::ws::generatePrivKey();
+   auto certServer = bs::network::ws::generateSelfSignedCert(privKeyServer);
+   auto pubKeyServer = bs::network::ws::publicKey(privKeyServer);
+   SPDLOG_LOGGER_DEBUG(StaticLogger::loggerPtr, "server public key: {}", bs::toHex(pubKeyServer));
+
+   auto privKeyClient = bs::network::ws::generatePrivKey();
+   auto certClient = bs::network::ws::generateSelfSignedCert(privKeyClient);
+   auto pubKeyClient = bs::network::ws::publicKey(privKeyClient);
+   SPDLOG_LOGGER_DEBUG(StaticLogger::loggerPtr, "client public key: {}", bs::toHex(pubKeyClient));
+
+   bool serverPubKeyValid = false;
+   bool clientPubKeyValid = false;
+
+   SslServerConnectionParams serverParams;
+   serverParams.useSsl = true;
+   serverParams.requireClientCert = true;
+   serverParams.cert = certServer;
+   serverParams.privKey = privKeyServer;
+   serverParams.verifyCallback = [&clientPubKeyValid, pubKeyClient](const std::string &pubKey) -> bool {
+      SPDLOG_LOGGER_DEBUG(StaticLogger::loggerPtr, "client public key: {}", bs::toHex(pubKey));
+      clientPubKeyValid = pubKeyClient == pubKey;
+      return true;
```

Comment:
I think the return value here should be `clientPubKeyValid` (and `serverPubKeyValid` below)

Reply:
Sure, fixed.